src/lib/libemval.js

```diff
@@ -467,33 +467,33 @@ var LibraryEmVal = {
     return result.done ? 0 : Emval.toHandle(result.value);
   },
 
-  _emval_coro_suspend__deps: ['$Emval', '_emval_coro_resume'],
-  _emval_coro_suspend: async (promiseHandle, awaiterPtr) => {
-    var result = await Emval.toValue(promiseHandle);
-    __emval_coro_resume(awaiterPtr, Emval.toHandle(result));
+  _emval_coro_suspend__deps: ['$Emval', '_emval_coro_resume',  '_emval_coro_reject'],
+  _emval_coro_suspend: (promiseHandle, awaiterPtr) => {
+    Emval.toValue(promiseHandle)
+      .then(result => __emval_coro_resume(awaiterPtr, Emval.toHandle(result)))
+      .catch(error => __emval_coro_reject(awaiterPtr, Emval.toHandle(error)));
   },
 
-  _emval_coro_make_promise__deps: ['$Emval', '__cxa_rethrow'],
+  _emval_coro_make_promise__deps: ['$Emval'],
   _emval_coro_make_promise: (resolveHandlePtr, rejectHandlePtr) => {
     return Emval.toHandle(new Promise((resolve, reject) => {
-      const rejectWithCurrentException = () => {
-        try {
-          // Use __cxa_rethrow which already has mechanism for generating
-          // user-friendly error message and stacktrace from C++ exception
-          // if EXCEPTION_STACK_TRACES is enabled and numeric exception
-          // with metadata optimised out otherwise.
-          ___cxa_rethrow();
-        } catch (e) {
-          // But catch it so that it rejects the promise instead of throwing
-          // in an unpredictable place during async execution.
-          reject(e);
-        }
-      };
-
       {{{ makeSetValue('resolveHandlePtr', '0', 'Emval.toHandle(resolve)', '*') }}};
-      {{{ makeSetValue('rejectHandlePtr', '0', 'Emval.toHandle(rejectWithCurrentException)', '*') }}};
+      {{{ makeSetValue('rejectHandlePtr', '0', 'Emval.toHandle(reject)', '*') }}};
     }));
   },
+
+  _emval_coro_val_from_current_cxa_exception__deps: ['$Emval', '__cxa_rethrow'],
+  _emval_coro_val_from_current_cxa_exception: () => {
+    try {
+      // Use __cxa_rethrow which already has mechanism for generating
+      // user-friendly error message and stacktrace from C++ exception
+      // if EXCEPTION_STACK_TRACES is enabled and numeric exception
+      // with metadata optimised out otherwise.
+      ___cxa_rethrow();
+    } catch (e) {
+      return Emval.toHandle(e);
+    }
```

Comment:
@aheejin does this look reasonable to you?  Or is the maybe a better way to do this?

Reply:
Do we wait for @aheejin's review or merge as-is? This piece of code was before this PR (I added it a while back because it was the only way to make it work that I found at the time), so I'm leaning towards let's merge it anyway.

Reply:
ping @aheejin